cupy/functional/piecewise.py

```diff
@@ -0,0 +1,80 @@
+import cupy
+
+from cupy import core
+from cupy.core.core import ndarray
+from cupy.core._reduction import ReductionKernel
+
+try:
+    from collections.abc import Callable
+except ImportError:
+    from collections import Callable
+
+_piecewise_krnl = ReductionKernel(
+    'S x1, T x2',
+    'U y',
+    'x1 ? x2 : NULL',
+    'b == NULL? a : b',
+    'y = a',
+    'NULL',
+    'piecewise'
+)
+
+
+def piecewise(x, condlist, funclist):
+    """
+    Evaluate a piecewise-defined function.
+
+    Args:
+    :param x: input domain
+    :param condlist: conditions list ( boolean arrays or boolean scalars)
+                     Each boolean array/ scalar corresponds to a function
+                     in funclist. Length of functionlist is equal to that
+                     of condlist. If one extra function is given, it is used
+                     as otherwise condition
+    :param funclist: list of scalar functions.
+
+    Returns:
+        cupy.ndarray: the result of calling the functions in funclist
+        on portions of x defined by condlist.
+
+    .. warning::
+
+        This function currently doesn't support callable functions
+        args and kw parameters are not supported
+
+    .. seealso:: :func:`numpy.piecewise`
+    """
+
+    if any(isinstance(item, Callable) for item in funclist):
+        raise ValueError('Callable functions are not supported')
```

Comment:
Please raise `NotImplementedError`.
```suggestion
        raise NotImplementedError('Callable functions are not supported currently')
```